### PR TITLE
refactor: Refactor CLI of Flank-scripts

### LINKED ISSUE #1567: Implement new flank-scripts commands structure and namings
According to the specification created in the result of #1545, implement the required changes.

### PR DESCRIPTION
**Pull request by piotradamczyk5**
Fixes #1567

This PR refactor CLI layer of Flank scripts as described in [command overview ](https://github.com/Flank/flank/blob/master/docs/flank-scripts/command_overview.md) and [SDD](https://github.com/Flank/flank/issues/1406)

1. Directories looks like: 
```bash
├── cli
│   ├── Main.kt
│   ├── assemble
│   │   ├── AssembleCommand.kt
│   │   ├── FlankCommand.kt
│   │   ├── GoCommand.kt
│   │   ├── android
│   │   │   ├── AndroidCommand.kt
│   │   │   └── AppCommand.kt
│   │   └── ios
│   │       ├── EarlGreyCommand.kt
│   │       ├── ExampleCommand.kt
│   │       ├── FlankExampleCommand.kt
│   │       ├── FtlCommand.kt
│   │       ├── GameLoopExampleCommand.kt
│   │       ├── IosCommand.kt
│   │       ├── RunFtlLocalCommand.kt
│   │       └── TestPlansExample.kt
│   ├── dependencies
│   │   ├── DependenciesCommand.kt
│   │   ├── InstallXcPrettyCommand.kt
│   │   ├── SetupIosEnvCommand.kt
│   │   ├── UniversalFrameworkCommand.kt
│   │   ├── UpdateBinariesCommand.kt
│   │   └── UpdateCommand.kt
│   ├── firebase
│   │   ├── CheckForSdkUpdatesCommand.kt
│   │   ├── FirebaseCommand.kt
│   │   ├── GenerateClientCommand.kt
│   │   └── UpdateApiCommand.kt
│   ├── github
│   │   ├── CopyIssuePropertiesCommand.kt
│   │   ├── DeleteOldTagCommand.kt
│   │   ├── DeleteReleaseCommand.kt
│   │   ├── GitHubCommand.kt
│   │   └── MakeReleaseCommand.kt
│   ├── integrationtests
│   │   ├── IntegrationTestsCommand.kt
│   │   └── ProcessResultCommand.kt
│   ├── linter
│   │   ├── ApplyToGitHooksCommand.kt
│   │   ├── ApplyToIdeCommand.kt
│   │   └── LinterCommand.kt
│   ├── release
│   │   ├── DeleteSnapshotCommand.kt
│   │   ├── GenerateReleaseNotesCommand.kt
│   │   ├── NextTagCommand.kt
│   │   ├── ReleaseCommand.kt
│   │   └── SyncWithMavenCentralCommand.kt
│   └── testartifacts
│       ├── DownloadCommand.kt
│       ├── LinkCommand.kt
│       ├── PrepareCommand.kt
│       ├── RemoveRemoteCommand.kt
│       ├── ResolveCommand.kt
│       ├── TestArtifactsCommand.kt
│       ├── UnzipCommand.kt
│       ├── UploadCommand.kt
│       └── ZipCommand.kt
```
2. Command are named and used like described in `docs/flank-scripts/command_overview.md`

## Checklist

- [x] Documented

### CODE REVIEW

**Review by adamfilipow92 — APPROVED** [review]
- [X] Code looks good
- [X] Build works

👍

File: docs/flank-scripts/README.md [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+  - Run script `buildFlankScripts.sh` in `flank-scripts/bash/` directory
+  - Run command `./gradlew clean flank-scripts:assemble flank-scripts:shadowJar` and manual copy file from `/flank-scripts/build/libs/flank-scripts.jar` to `flank-scripts/bash/`
+  - You could always run/build it from Intellij IDEA 
+
+### Usage
+
+Run the script with arguments
+`flankScripts <command group> [<subgroup>] <command name> [<arguments>]`
+
+If you need help with available commands or arguments you could always use option `--help`
```
Comment (pawelpasterz) [inline docs/flank-scripts/README.md:18]:
```suggestion
If you need help with available commands or arguments you could always use the option `--help`
```

File: docs/flank-scripts/README.md [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+  - `next_tag` - Get tag for next release
+  - `sync_with_maven_central` - Sync artifact's repository with Maven central
+
+___
+
+- `test_artifacts` - Group of commands for artifacts management
+  - `download` - Download test artifacts zip asset to test_artifacts directory.
+  - `link` - Create symbolic link to under test_runner/src/test/kotlin/ftl/fixtures/tmp to
+    test_artifacts/{branchName}.
+  - `prepare`- Creates fresh copy of test artifacts for current working branch, basing on existing one.
```
Comment (pawelpasterz) [inline docs/flank-scripts/README.md:87]:
```suggestion
  - `prepare`- Creates a fresh copy of test artifacts for the current working branch, basing on an existing one.
```

File: docs/flank-scripts/README.md [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+
+### Arguments
+
+To show applicable arguments for command use `--help` or `-h` options:
+`flankScripts <command group> [<subgroup>] <command name> --help`
+or
+`flankScripts <command group> [<subgroup>] <command name> -h`
+
+## Testing
+To test your script with different settings use `flank-debug.properties` file. Uncomment and replace with desired values.
```
Comment (pawelpasterz) [inline docs/flank-scripts/README.md:105]:
```suggestion
To test your script with different settings use the `flank-debug.properties` file. Uncomment and replace with desired values.
```

File: docs/flank-scripts/README.md [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+Properties are skipped by git and should not be attached to a commit. Note, `test` task ignores your own properties and will use the default.
```
Comment (pawelpasterz) [inline docs/flank-scripts/README.md:106]:
```suggestion
Properties are skipped by git and should not be attached to a commit. Note, the `test` task ignores your own properties and will use the default.
```

File: docs/flank-scripts/README.md (hunk 2 above)
Comment (piotradamczyk5) [reply docs/flank-scripts/README.md:87]: fixed

**Review by pawelpasterz — APPROVED** [review]

**Pull request merged**